conda-store-server/conda_store_server/orm.py

```diff
@@ -234,15 +257,29 @@ def build_key(self):
         The build key should be a key that allows for the environment
         build to be easily identified and found in the database.
         """
-        datetime_format = "%Y%m%d-%H%M%S-%f"
-        return f"{self.specification.sha256}-{self.scheduled_on.strftime(datetime_format)}-{self.id}-{self.specification.name}"
+        if self.build_key_version == 1:
+            datetime_format = "%Y%m%d-%H%M%S-%f"
+            return f"{self.specification.sha256}-{self.scheduled_on.strftime(datetime_format)}-{self.id}-{self.specification.name}"
+        elif self.build_key_version == 2:
+            hash = self.specification.sha256[:_BUILD_KEY_V2_HASH_SIZE]
+            timestamp = int(self.scheduled_on.timestamp())
+            id = self.id
+            name = self.specification.name
+            return f"{hash}-{timestamp}-{id}-{name}"
```

Comment:
It feels that these should be better encoded in separate callables for clarity. Something like a dictionary of versions -> callables, that implement each format cleanly. This would also double as the catalog of available versions for validation and error reporting.